spec/requests/objects_controller_file_spec.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ObjectsController, type: :request do
+  let(:prefixed_druid) { 'druid:bj102hs9687' }
+  let(:bare_druid) { 'bj102hs9687' }
+  let(:pres_obj) { create(:preserved_object) }
+
+  describe 'GET #file' do
+    context 'when object exists' do
+      context 'when druid is prefixed' do
+        context 'when no version param specified' do
+          context 'when file is present' do
+            context 'when type is manifest' do
```

Comment:
I wonder whether you'd consider flattening this a bit. Perhaps by setting more variables above to assume happy path, and then varying one unhappy path per `context` block. That might test less interaction than you have now, but the code you're testing doesn't look too complicated, so I'm wondering whether the tests need to be this nested. You've spent more time wrapping your brain around this, though, and if you think this is a simplistic suggestion... you're probably right. :smile:

Reply:
addressed with latest changes.